sdk/search/azure-search/src/test/java/com/azure/search/SearchTestBase.java

```diff
@@ -0,0 +1,427 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.search;
+
+import com.azure.search.models.DataType;
+import com.azure.search.models.FacetResult;
+import com.azure.search.models.Field;
+import com.azure.search.models.Index;
+import com.azure.search.models.QueryType;
+import com.azure.search.models.RangeFacetResult;
+import com.azure.search.models.RequestOptions;
+import com.azure.search.models.SearchOptions;
+import com.azure.search.models.SearchResult;
+import com.azure.search.models.SynonymMap;
+import com.azure.search.models.ValueFacetResult;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import org.junit.Assert;
+import org.junit.jupiter.api.Test;
+import org.unitils.reflectionassert.ReflectionComparatorMode;
+
+import java.io.IOException;
+import java.text.ParseException;
+import java.time.OffsetDateTime;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.unitils.reflectionassert.ReflectionAssert.assertReflectionEquals;
+
+/**
+ * Abstract base class for all Search API tests
+ */
+public abstract class SearchTestBase extends SearchIndexClientTestBase {
+
+    static final String HOTELS_INDEX_NAME = "hotels";
+    static final String HOTELS_DATA_JSON = "HotelsDataArray.json";
+    static final String HOTELS_DATA_JSON_WITHOUT_FR_DESCRIPTION = "HotelsDataArrayWithoutFr.json";
+    private static final String SEARCH_SCORE_FIELD = "@search.score";
+
+    protected List<Map<String, Object>> hotels;
+
+    List<Map<String, Object>> createHotelsList(int count) {
+        List<Map<String, Object>> documents = new ArrayList<>();
+        for (int i = 1; i <= count; i++) {
+            Map<String, Object> doc = new HashMap<>();
+
+            doc.put("HotelId", Integer.toString(i));
+            doc.put("HotelName", "Hotel" + i);
+            doc.put("Description", "Desc" + i);
+            doc.put("Description_fr", "Desc_fr" + i);
+            doc.put("Category", "Catg" + i);
+            doc.put("Tags", Collections.singletonList("tag" + i));
+            doc.put("ParkingIncluded", false);
+            doc.put("SmokingAllowed", false);
+            doc.put("LastRenovationDate", OffsetDateTime.parse("2010-06-27T00:00:00Z"));
+            doc.put("Rating", i);
+
+            documents.add(doc);
+        }
+        return documents;
+    }
+
+    /**
+     * Drop fields that shouldn't be in the returned object
+     *
+     * @param map the map to drop items from
+     * @return the new map
+     */
+    static Map<String, Object> dropUnnecessaryFields(Map<String, Object> map) {
```

Comment:
@navalev Why is this necessary? Looking at the generated code for `SearchResult`, it looks like the score is already deserialized into a separate field, so why would it also end up in the map? And if it does, why not other annotations like highlights? Is this working around a bug in azure.core?

Reply:
@brjohnstmsft The method only removed the @search.score field. Since the result of the search query retuns a map of objects (and not get deserialized into an object in our case), the value of @search.score also gets into this map. To make sure we compare only the document in the test, this value if cleaned up.

Reply:
@navalev Ok, that sounds like a bug in AutoRest to me, because `@search.score` is already modeled as a separate property in the Swagger. `additionalProperties` is only supposed to contain the values that aren't mapped to other properties of the type. We will follow up on this and make sure it's fixed before we ship, because having annotations like `@search.score` in the data returned to customers is unacceptable.